lib/Element.js

```diff
@@ -279,13 +277,24 @@ Element.prototype.toReact = function (index) {
     }
   }))
 
-  return React.createElement(this.nodeName, props, this.text || this.children.map(function (el, i) {
-    if (el instanceof Element) {
-      return el.toReact(i)
-    } else {
+  var children
+
+  if (this.text) {
+    children = this.text
+  } else if (this.children.length === 1) {
+    var el = this.children[0]
+    children = el instanceof Element ? el.toReact() : el
+  } else if (this.children.length > 1) {
+    children = this.children.map(function (el) {
+      if (el instanceof Element) {
+        return el.toReact(el.key)
+      }
+
       return el
-    }
-  }))
+    })
+  }
+
+  return React.createElement(this.nodeName, props, children)
```

Comment:
Well, it's hoisted regardless. I would move it to another file and require it but it becomes a cyclic dependency :(

You could add it above if you want, I don't mind.